File: estonian-courier-sim/app/src/main/java/ee/kuller/app/ui/screens/PhrasebookScreen.kt
package ee.kuller.app.ui.screens

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import ee.kuller.app.data.Content
import ee.kuller.app.ui.SectionTitle
import ee.kuller.app.ui.WordRow
import ee.kuller.app.util.rememberSpeaker

/** Разговорник: готовые фразы курьера по темам, с озвучкой. */
@Composable
fun PhrasebookScreen() {
    val speaker = rememberSpeaker()
    // В разговорнике сначала самые нужные курьеру темы
    val order = listOf("phrase", "greet", "dir", "num", "food", "drink")
    val cats = order.mapNotNull { id -> Content.categories.firstOrNull { it.id == id } }

    LazyColumn(
        Modifier.fillMaxSize().padding(horizontal = 16.dp),
        verticalArrangement = Arrangement.spacedBy(2.dp)
    ) {
        item {
            Text(
                "Vestmik · Разговорник",
                fontWeight = FontWeight.Bold,
                fontSize = 24.sp,
                color = MaterialTheme.colorScheme.onBackground,
                modifier = Modifier.padding(top = 16.dp, bottom = 4.dp)
            )
            Text(
                "Нажми 🔊, чтобы услышать произношение",
                fontSize = 13.sp,
                color = MaterialTheme.colorScheme.onBackground.copy(alpha = 0.7f),
                modifier = Modifier.padding(bottom = 8.dp)
            )
        }
        cats.forEach { cat ->
            item { SectionTitle("${cat.emoji}  ${cat.titleEt} · ${cat.titleRu}") }
            items(cat.words) { word ->
                WordRow(
                    et = word.et,
                    ru = word.ru,
                    example = word.example,
                    onSpeak = { speaker.speak(if (word.example.isNotBlank()) word.example else word.et) }
                )
            }
        }
        item { Text("", Modifier.padding(8.dp)) }
    }
}

File: kuller_flutter/lib/screens/phrasebook_screen.dart
import 'package:flutter/material.dart';

import '../content.dart';
import '../game_vm.dart';
import '../widgets.dart';

/// Разговорник: готовые фразы курьера по темам, с озвучкой.
class PhrasebookScreen extends StatelessWidget {
  final GameViewModel vm;
  const PhrasebookScreen({super.key, required this.vm});

  @override
  Widget build(BuildContext context) {
    final scheme = Theme.of(context).colorScheme;
    // В разговорнике сначала самые нужные курьеру темы
    const order = ['phrase', 'greet', 'money', 'foodq', 'dir', 'num', 'food', 'drink'];
    final cats = [
      for (final id in order)
        Content.categories.firstWhere((c) => c.id == id),
    ];

    return ListView(
      padding: const EdgeInsets.symmetric(horizontal: 16),
      children: [
        Padding(
          padding: const EdgeInsets.only(top: 16, bottom: 4),
          child: Text('Vestmik · Разговорник',
              style: TextStyle(
                  fontWeight: FontWeight.bold,
                  fontSize: 24,
                  color: scheme.onSurface)),
        ),
        Padding(
          padding: const EdgeInsets.only(bottom: 8),
          child: Text('Нажми 🔊, чтобы услышать произношение',
              style: TextStyle(
                  fontSize: 13, color: scheme.onSurface.withOpacity(0.7))),
        ),
        for (final cat in cats) ...[
          SectionTitle('${cat.emoji}  ${cat.titleEt} · ${cat.titleRu}'),
          for (final word in cat.words)
            WordRow(
              et: word.et,
              ru: word.ru,
              example: word.example,
              onSpeak: () =>
                  vm.tts.speak(word.example.isNotEmpty ? word.example : word.et),
            ),
        ],
        const SizedBox(height: 16),
      ],
    );
  }
}
